// Conformance tests for all scalar types.
syntax = "proto3";

package conformance;

// All scalar types in a single message for comprehensive testing.
message Scalars {
  int32 val_int32 = 1;
  int64 val_int64 = 2;
  uint32 val_uint32 = 3;
  uint64 val_uint64 = 4;
  sint32 val_sint32 = 5;
  sint64 val_sint64 = 6;
  bool val_bool = 7;
  fixed32 val_fixed32 = 8;
  sfixed32 val_sfixed32 = 9;
  fixed64 val_fixed64 = 10;
  sfixed64 val_sfixed64 = 11;
  float val_float = 12;
  double val_double = 13;
  string val_string = 14;
  bytes val_bytes = 15;
}

// Individual scalar messages for isolated testing.
message Int32Value { int32 value = 1; }
message Int64Value { int64 value = 1; }
message Uint32Value { uint32 value = 1; }
message Uint64Value { uint64 value = 1; }
message Sint32Value { sint32 value = 1; }
message Sint64Value { sint64 value = 1; }
message BoolValue { bool value = 1; }
message Fixed32Value { fixed32 value = 1; }
message Sfixed32Value { sfixed32 value = 1; }
message Fixed64Value { fixed64 value = 1; }
message Sfixed64Value { sfixed64 value = 1; }
message FloatValue { float value = 1; }
message DoubleValue { double value = 1; }
message StringValue { string value = 1; }
message BytesValue { bytes value = 1; }
